### PR TITLE
[SYCL][E2E] Remove -c flags on run lines that use -fsycl-device-only

### PR DESCRIPTION
**Pull request by ayylol**
The `-c` flag is not used when used in conjunction with the `-fsycl-device-only` flag. This patch removes uses of `-c` in such situations.

This caused failures in internal testing, because the closed source compiler reports this as an `unused-command-line-argument` warning, however syclos does not report this.

### CODE REVIEW

**Review by steffenlarsen — APPROVED** [review]
🚀

**Comment by againull** [comment]
@intel/dpcpp-tools-reviewers  friendly ping

**Comment by aelovikov-intel** [comment]
> however syclos does not report this.

I think this needs a better wording

**Comment by AlexeySachkov** [comment]
This patch was merged without required approvals and even though I'm fine with that in this particular case (because it is a small NFC change into tests) I think it worth leaving a comment that gatekeepers decided to force-merge a patch without waiting for all code owners just to make it clear for everyone.

**Pull request merged**
